e2e/cloud-plugins-suite/azure-monitor.spec.ts

```diff
@@ -45,6 +46,7 @@ function provisionAzureMonitorDatasources(datasources: AzureMonitorProvision[])
       e2eSelectors.configEditor.defaultSubscription.input().find('input').type('datasources{enter}');
     },
     expectedAlertMessage: 'Successfully connected to all Azure Monitor endpoints',
+    timeout: 10000,
   });
```

Comment:
another possibility is adding here a try-catch if we know this is the specific step that may fail. That way we click the "Save" button twice rather than creating a new data source.

Reply:
Try-catch statements don't work in Cypress :/ Cypress is designed to prevent you from retrying specific steps (that was my initial attempt, see details [here](https://docs.cypress.io/guides/core-concepts/conditional-testing#Error-Recovery)). I agree with all of your comments above, particularly about hiding legitimate errors. I'll have more of a think on this and maybe implement some uncaught exception handling.